system/wymeditor/plugins/image_upload/jquery.wymeditor.image_upload.js

```diff
@@ -45,12 +45,37 @@ WYMeditor.editor.prototype.image_upload = function() {
 				} else {
 					jQuery(".wym_src", doc).val(response.thumbUrl);
 					jQuery(".wym_alt", doc).val(response.original_filename);
+					jQuery(".wym_attribution", doc).val(response.attribution);
 				}
 				jQuery("form#image_upload_form .submit", doc).val(oldSubmitLabel);
 			}
 		})
 	};
 
+	d.submitHandler = function (wDialog) {
+		var wym = this,
+			options = wym._options,
+			imgAttrs,
+			selectedImage = wym.getSelectedImage();
+
+		imgAttrs = {
+			src: jQuery(options.srcSelector, wDialog.document).val(),
+			title: jQuery(options.titleSelector, wDialog.document).val(),
+			alt: jQuery(options.altSelector, wDialog.document).val(),
+			longdesc: jQuery('.wym_attribution', wDialog.document).val()
```

Comment:
This seems weird: You enter an attribution, which is then saved as a "long description", and displayed as a caption, which seem three distinct things that I think we should not mix here? Maybe this is also something that @giplt could comment on?

Reply:
I've implemented it like it was requested by @tammoterhark, I think he had a clear vision about this.

Reply:
updated to "attribution"